src/components/forms/ComboBox/ComboBox.tsx

```diff
@@ -78,372 +85,391 @@ const Input = ({
 
-  // If the focused element (activeElement) is outside of the combo box,
-  // make sure the focusMode is BLUR
-  useEffect(() => {
-    if (state.focusMode !== FocusMode.None) {
-      if (!containerRef.current?.contains(window.document.activeElement)) {
-        dispatch({
-          type: ActionTypes.BLUR,
-        })
+    const [state, dispatch] = useComboBox(
+      initialState,
+      options,
+      disableFiltering,
+      filter
+    )
+
+    const containerRef = useRef<HTMLDivElement>(null)
+    const focusedItemRef = useRef<HTMLLIElement>(null)
+
+    useEffect(() => {
+      onChange && onChange(state.selectedOption?.value || undefined)
+    }, [state.selectedOption])
+
+    useEffect(() => {
+      if (
+        state.focusMode === FocusMode.Item &&
+        state.focusedOption &&
+        focusedItemRef.current
+      ) {
+        focusedItemRef.current.focus()
       }
-    }
-  }, [state.focusMode])
-
-  const handleInputKeyDown = (event: KeyboardEvent): void => {
-    if (event.key === 'Escape') {
-      dispatch({ type: ActionTypes.CLOSE_LIST })
-    } else if (event.key === 'ArrowDown' || event.key == 'Down') {
-      event.preventDefault()
-      dispatch({
-        type: ActionTypes.FOCUS_OPTION,
-        option:
-          state.selectedOption ||
-          state.focusedOption ||
-          state.filteredOptions[0],
-      })
-    } else if (event.key === 'Tab') {
-      // Clear button is not visible in this case so manually handle focus
-      if (state.isOpen && !state.selectedOption) {
-        // If there are filtered options, prevent default
-        // If there are "No Results Found", tab over to prevent a keyboard trap
-        const optionToFocus = disableFiltering
-          ? state.focusedOption
-          : state.selectedOption || state.focusedOption
-        if (optionToFocus) {
-          event.preventDefault()
-          dispatch({
-            type: ActionTypes.FOCUS_OPTION,
-            option: optionToFocus,
-          })
-        } else {
+    }, [state.focusMode, state.focusedOption])
+
+    // When opened, the list should scroll to the closest match
+    useEffect(() => {
+      if (
+        state.isOpen &&
+        state.focusedOption &&
+        focusedItemRef.current &&
+        state.focusMode === FocusMode.Input
+      ) {
+        focusedItemRef.current.scrollIntoView(false)
+      }
+    }, [state.isOpen, state.focusedOption])
+
+    // If the focused element (activeElement) is outside of the combo box,
+    // make sure the focusMode is BLUR
+    useEffect(() => {
+      if (state.focusMode !== FocusMode.None) {
+        if (!containerRef.current?.contains(window.document.activeElement)) {
           dispatch({
             type: ActionTypes.BLUR,
           })
         }
       }
-
-      if (!state.isOpen && state.selectedOption) {
+    }, [state.focusMode])
+
+    useImperativeHandle(
+      ref,
+      () => ({
+        clearSelection: (): void => dispatch({ type: ActionTypes.CLEAR }),
```

Comment:
I'm not sure if this is a bad thing or not, but it may be worth aligning on:

`ActionTypes.CLEAR` gives focus to the input. Should that be the default behavior here? I could see dragging focus to the input being problematic under certain circumstances (like a form where values are programmatically pre-filled and cleared as a user is progressing through the form. The clear action would pull them to a different section of the form potentially)

Reply:
That's a good point, I hadn't really fully considered the implications of just glomming onto that existing action.

Reply:
Updated to use a new ActionType.

Reply:
Excellent. This feels in line with what I would expect as a consumer.